ndk/src/surface_texture.rs

```diff
@@ -0,0 +1,112 @@
+//! Bindings for [`ffi::ASurfaceTexture`]
+#[cfg(feature = "api-level-28")]
+use crate::native_window::NativeWindow;
+#[cfg(feature = "api-level-28")]
+use jni_sys::{jobject, JNIEnv};
+use std::ptr::NonNull;
+use thiserror::Error;
+
+#[derive(Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct SurfaceTexture {
+    ptr: NonNull<ffi::ASurfaceTexture>,
+}
+
+unsafe impl Send for SurfaceTexture {}
+
+#[derive(Debug, Error)]
+pub struct PosixError(pub i32);
+
+impl std::fmt::Display for PosixError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "Posix Error: {}", self.0)
```

Comment:
Good enough for now, we can stringify these later.

Since more Android API (to be added in the future) shares these error codes I think we should move this type to a separate module though.